File: CoreServer/datamodel/server/datamodelloaderbase.cpp
#include "datamodelloaderbase.h"

DatamodelLoaderBase::DatamodelLoaderBase(QObject *parent) : Identifyable("DatamodelLoaderBase"), QObject(parent)
{

}

LogCat::LOGCAT DatamodelLoaderBase::logCat() {
    return LogCat::DATAMODEL;
}

File: AndroidApp/OshCore/src/main/java/com/osh/datamodel/DatamodelLoaderBase.java
package com.osh.datamodel;

import com.osh.Identifyable;

public abstract class DatamodelLoaderBase extends Identifyable {

	public DatamodelLoaderBase(String id) {
		super(id);
	}
	
    public abstract DatamodelBase load();

    public abstract void save(DatamodelBase datamodel);

    public abstract String typeName();


}
